src/algebra/big_operators/fin.lean

```diff
@@ -161,6 +161,16 @@ end partial_prod
 
 end fin
 
+namespace vector
+
+variables {n : ℕ} (a : α) (v : vector α n)
+
+lemma sum_filter_eq_nth_eq_count [decidable_eq α] :
+  ∑ i in univ.filter (λ i, a = v.nth i), 1 = v.to_list.count a :=
+trans (by rw [sum_const, nsmul_one, nat.cast_id]) (fin.card_filter_univ_eq_nth_eq_count a v)
```

Comment:
I don't think this lemma is worth having; it can be proved with just two rewrites, and the first one should be obvious:
```lean
by rw [←card_eq_sum_ones, fin.card_filter_univ_eq_nth_eq_count a v]
```